Update Babel and Jest

fkloes opened the pull request:
# Description

This pull request updates all `Babel` related dependencies to the latest stable versions for improved compatibility, performance, and tooling support.

Additionally it updates `Jest` to the latest version that's compatible with the current Shopgate infrastructure.

## Type of change

- [ ] Bug Fix :bug: (non-breaking change which fixes an issue)
- [ ] Enhancement :rocket: (non-breaking change which adds functionality)
- [ ] Breaking Change :boom: (fix or feature that would cause existing functionality to not work as expected)
- [ ] Polish :nail_care: (Just some cleanups)
- [x] Internal :house: Only relates to internal processes.

## How to test it
Since the Jest update to version 29 introduced a new snapshot format, every snapshot had to be updated. I would recommend to filter snapshots from the changed files.

fkloes committed:
CURB-4562 Updated babel to latest versions

fkloes committed:
CURB-4562 Fixed some unit tests after babel update

fkloes committed:
CURB-4562 Fixed unit tests

fkloes committed:
CURB-4562 Updated babel.config files

- added browser targets

fkloes committed:
CURB-4562 Dropped no-restricted-syntax eslint rule for for of loops

- not valid anymore with modern babel setup

fkloes committed:
CURB-4562 Added dist folders to eslintignore list

fkloes committed:
CURB-4562 Removed unused package

fkloes committed:
CURB-4562 Updated Jest to v25

fkloes committed:
CURB-4562 Improved jest test

fkloes committed:
CURB-4562 Improved some tests

fkloes committed:
Merge branch 'CURB-5489-update-webpack' into CURB-4562-update-babel

# Conflicts:
#	package.json
#	themes/theme-gmd/babel.config.js
#	themes/theme-gmd/package.json
#	themes/theme-ios11/babel.config.js
#	themes/theme-ios11/package.json
#	yarn.lock

fkloes committed:
Merge branch 'CURB-5489-update-webpack' into CURB-4562-update-babel

# Conflicts:
#	yarn.lock

fkloes committed:
CURB-4562 Update Jest to v28

fkloes committed:
CURB-4562 Added option to skip long running unit tests and removed unused packages

fkloes committed:
CURB-4562 Removed isomorphic-fetch module

- fetch is supported by all modern browsers

fkloes committed:
CURB-4562 Removed unused babel plugins

fkloes committed:
CURB-4562 Refactored tests that imported JSDom and removed JSDom package

fkloes committed:
CURB-4562 Small improvement

fkloes committed:
CURB-4562 Refactored some tracking-core tests to jest

fkloes committed:
CURB-4562 Small improvements

fkloes committed:
CURB-4562 Refactored tracking-core tests to jest and removed mocha dependencies

fkloes committed:
CURB-4562 Updated Jest to version 29

fkloes committed:
CURB-4562 Removed package.lock from theme gmd

fkloes committed:
CURB-4562 Increased speed of some test

fkloes committed:
CURB-4562 Cared about vulnerable packages

fkloes committed:
CURB-4562 Removed package-lock from theme ios

fkloes committed:
CURB-4562 Added .browserslistrc to project

- guarantees shared build targets between babel and webpack

fkloes committed:
Merge branch 'CURB-5489-update-webpack' into CURB-4562-update-babel

# Conflicts:
#	extensions/@shopgate-product-reviews/frontend/package.json
#	libraries/tracking-core/package.json

fkloes committed:
Merge branch 'CURB-5489-update-webpack' into CURB-4562-update-babel

fkloes committed:
Merge branch '2025-tooling-update' into CURB-4562-update-babel

# Conflicts:
#	extensions/@shopgate-product-reviews/frontend/package.json
#	libraries/tracking-core/package.json

AylinUenal commented on libraries/tracking-core/tests/Core.spec.js:
```diff
@@ -94,15 +91,14 @@ describe('Core', () => {
+    const options = { trackerName: 'mock' };
     const expectedObject = { expected: 'data' };
 
+    // initial call as in original test
```
remove comment. original test is gone now

fkloes replied:
removed

fkloes committed:
Merge branch '2025-tooling-update' into CURB-4562-update-babel

# Conflicts:
#	extensions/@shopgate-product-reviews/frontend/package.json
#	libraries/tracking-core/package.json

AylinUenal commented:
I get this error when deleting or renaming a custom favorites list, can be reproduced every time:
<img width="1289" height="1120" alt="Screenshot 2025-12-10 at 15 13 31" src="https://github.com/user-attachments/assets/a0627187-e4b6-4455-98fa-46fe40cb65f2" />

AylinUenal commented:
when i enter the PDP of a product with variants, often the variant picker doesn't show up and I get this error (tested with shop 31331 on iphone 15 plus iOS 17):
<img width="1164" height="649" alt="Screenshot 2025-12-16 at 16 04 01" src="https://github.com/user-attachments/assets/1949db6b-793b-43aa-8a91-25049278bbfd" />

fkloes commented on libraries/common/collections/media-providers/YouTube.spec.js:
```diff
@@ -19,25 +19,35 @@ jest.mock('@shopgate/engage/core/helpers', () => ({
- * Creates a mocked document with iframes.
- * @param {Array} srcs A list of video URLs.
- * @return {Object}
+ * Creates a mocked document with iframes (parity with your original helper).
```
removed

fkloes committed:
CURB-4562 Updated some JSDoc

fkloes committed:
Merge branch '2025-tooling-update' into CURB-4562-update-babel

fkloes commented:
Comment 1:
I'll prepare an additional ticket to fix the issue in the favourites list. I don't think that it's cause by the babel update, but by the webpack update. Pull request: https://github.com/shopgate/pwa/pull/1440

Comment 2:
I was able to reproduce the issue when `ext-recently-viewed-products` is attached. The error came from [this function](https://github.com/shopgate/ext-recently-viewed-products/blob/e2f5da0f87a56433328dfea2c795bc075fd6f7ab/frontend/actions/index.js#L51) that might spread `undefined` into the array which is invalid for native spread. I've added `@babel/plugin-transform-spread` back to the babel-config, so such code is transpiled again. The plugin was removed previously because all of our supported browsers actually support spread syntax natively.

fkloes committed:
CURB-5649 Refactored internal ConnectedReactPortal component

- now uses createPortal from react-dom instead of the react-portal package

fkloes committed:
CURB-5649 Fixed react warning when favorites list dialog closes

fkloes committed:
CURB-4562 Re-added some babel plugins

- added @babel/plugin-transform-spread since extensions might have code that's not compatible with native spread
- added @babel/plugin-transform-classes since @babel/plugin-transform-spread doesn't work with native classes

Commit:
Merge branch 'CURB-4562-update-babel' into CURB-5649-refactor-connected-react-portal-component

Commit:
CURB-5649 update jsdoc

AylinUenal committed:
Merge pull request #1440 from shopgate/CURB-5649-refactor-connected-react-portal-component

Refactored internal ConnectedReactPortal component
